frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/sc-node-list.tsx

```diff
@@ -17,11 +18,16 @@ import '../ocs-install.scss';
 
 const getRows: GetRows = ({ componentProps, customData }) => {
   const { data } = componentProps;
-  const { nodes, setNodes, filteredNodes } = customData;
+  const { nodes, setNodes, filteredNodes, isHostNameSame } = customData;
+  let filteredData: NodeKind[] = [];
 
   const nodeList = filteredNodes ?? data.map(getName);
-  const hostNames = createMapForHostNames(data);
-  const filteredData = data.filter((node) => nodeList.includes(hostNames[getName(node)]));
+  if (isHostNameSame) {
+    filteredData = data.filter((node) => nodeList.includes(getName(node)));
+  } else {
+    const hostNames = createMapForHostNames(data);
+    filteredData = data.filter((node) => nodeList.includes(hostNames[getName(node)]));
+  }
```

Comment:
Wont any of these work ?
```suggestion
  const filteredData = data.filter(
    (node: NodeKind) =>
      nodeList.includes(getName(node)) ||
      nodeList.includes(node.metadata.labels?.['kubernetes.io/hostname']),
  );
```
OR

Pass the maintained state from pre-step about the "filteredNodes"

Reply:
We already have the function - `createMapForHostNames`, which checks for the hostname and hence no need to do the similar check here.
Also this component is reused at multiple places, hence it make sense to do the check here, instead of calculating the same from all different components.

Reply:
I am okay If you want to pass prop but I think in any scenario, you dont need to `createMapForHostNames`, you can directly do,
```
nodeList.includes(node.metadata.labels?.['kubernetes.io/hostname'])
```
 Its just extra computation.